libs/wcm/hooks/useConnectionEventsManager.js

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useContext } from 'react';
+import { client } from '@libs/wcm/utils/connectionCreator';
+import ConnectionContext from '../context/connectionContext';
+import { EVENTS } from '../constants/lifeCycle';
+
+const useWalletConnectEventsManager = () => {
+  const {
+    pushEvent, disconnect, session, setSession,
+  } = useContext(ConnectionContext);
+
+  const onSessionRequest = useCallback(async (event) => {
+    const request = client.session.get(event.topic);
+
+    setSession({ ...session, request });
+  }, []);
+
+  const onSessionDelete = useCallback((event) => {
+    disconnect(event.topic);
+  }, []);
+
+  const eventHandler = useCallback((name, meta) => {
+    pushEvent({ name, meta });
+
+    switch (name) {
+      case EVENTS.SESSION_DELETE:
+        onSessionDelete(meta);
+        break;
+      case EVENTS.SESSION_REQUEST:
+        onSessionRequest(meta);
+        break;
+      default:
+        break;
+    }
```

Comment:
I don't know how many of these events can come handy for other apps, because I don't have a clear picture of all apps in the Lisk ecosystem. Being it a platform made for everyone, I assume there will be more versatile use-cases, and more events to listen to. Switch case seemed a better placeholder. Nevertheless, if you think we should simplify it, I can change it.

Reply:
For the scope we know let's keep it simple, maybe when we have to extend we can update the implementation 👍🏼